environment.yml

```diff
@@ -2,60 +2,74 @@ name: guide
 channels:
   - jaimergp/label/unsupported-cudatoolkit-shim
   - conda-forge
-  - plotly
   - defaults
 dependencies:
+  # mdanalysis dependencies
   - cython
-  - distributed==2021.1.0
-  - dask==2021.1.0
-  - python-graphviz
-  - pytest
-  - decorator
-  - docutils
-  - graphviz
-  - ipywidgets<8.0.0
-  - ipython
-  - ipygany
-  - jedi
-  - jinja2
-  - jupyter_contrib_nbextensions
   - matplotlib
-  - nbclient
-  - nbconvert
-  - nbformat
-  - nbsphinx
   - networkx
-  - nglview
-  - notebook
   - numpy
   - packaging
   - parmed
-  - parso
-  - pip
-  - plotly
-  - pybtex
-  - python=3.9
-  - pyvista
-  - scikit-image
+  - pyedr>=0.7
   - scikit-learn>=1.1.2
+  - threadpoolctl
+  # - decorator # not used?
+  # building documentation
+  - docutils
+  - pandoc
+  - pybtex
   - sphinx
   - sphinxcontrib-bibtex
-  - threadpoolctl
-  - tabulate
-  - widgetsnbextension
-  - ipycytoscape>=1.3.0
-  - seaborn>=0.12.1
+  # - jedi # not used?
+  # - jinja2 # not used?
+  # - nbclient not used?
+  # - parso not used?
+  - pip
+  - python=3.9
+  # - widgetsnbextension not used?
+  # - ipycytoscape>=1.3.0 not used?
   - openmm>=7.0.0
-  - pandoc
-  - moviepy
-  - ipyvtklink
+  # - ipyvtklink not used?
+  # used by custom_parallel_analysis.ipynb
+  - graphviz
+  - python-graphviz
   - pip:
+      # used in multiple notebooks
+      - notebook<7
+      - jupyter_contrib_nbextensions
+      - nbconvert
+      - nbformat
+      - nglview
+      - seaborn>=0.12.1
+      # used in pairwise_rmsd.ipynb
+      - plotly
+      # used in custom_parallel_analysis.ipynb
+      - distributed==2021.1.0
+      - dask==2021.1.0
+      # used in density_analysis.ipynb
+      - ipygany
+      # used in density_analysis.ipynb
+      - pyvista
+      # used in density_analysis.ipynb
+      - scikit-image
+      # used in aligning_trajectory.ipynb
+      - moviepy
+      # used in conf.py
+      - ipython
       - sphinx_rtd_theme
       - msmb_theme
       - sphinx-sitemap
+      - ipywidgets<8.0.0
+      - nbsphinx
+      # used for notebook testing
       - nbval
+      # building documentation
+      - tabulate
+      # tests for this package
       - pre-commit
-      - pyedr>=0.7
+      - pytest
```

Comment:
It's not just that but yeah I see your point. Just wanted to push things around, it's an exploratory PR. It ideally shouldn't matter if the package is from pip vs conda channel. 

For a user guide package, there are many deps and many were found to be obsolete (the build still passes). From things under `pip` we can more easily migrate to subpackages in pyproject.toml (where people who just want to build don't need to install syrup, or some other tests framework).

Reply:
It's just quite hard for a new comer to understand "why all these deps? what is each needed for?"

The more we can put into pip, the more understandable things will get in my view – it'll surface things that are "conda-only" and that will be communicated explicitly. `openmm` for example is only available in conda, same for graphviz, and a few others. it's not clear that's the case until you start messing about ;-)

Reply:
I'm not sure the benefit of communicating what is only on conda vs what is only on pip is worth the additional [environment fragility](https://www.anaconda.com/blog/using-pip-in-a-conda-environment). Perhaps I'm not fully understanding, since you've also left a number of packages in the conda part (e.g. matplotlib) that are available on pypi. I think we could turn this around and communicate "these packages are *not* available on conda-forge" with the `pip` section, which as I understand it is the standard way to structure these environment files.

> It's just quite hard for a new comer to understand "why all these deps? what is each needed for?"

I think the comments you've put in separating the sections into mda/docs sections are better at communicating this, and go a long way to making it much clearer!

> The last cherry is that the env is now much slimmer: 1.8 GB vs 2.6 GB. But that's a bonus objective, just improving on the margins.

I assume much of this is eliminating CUDA and other unnecessary packages?

Reply:
That's a good point @lilyminium I wasn't super consistent moving everything under pip. 

Okay, let's take a step back here – I concede that this PR is trying to do too much – I'll undo the these pip/conda reshuffles, keep the comments and ditch the packages we no longer need